inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/conf/ConfigJetty.java

```diff
@@ -79,10 +80,22 @@ public void storeJobConf(JobProfile jobProfile) {
             if (jobProfile.hasKey(JOB_TRIGGER)) {
                 triggerManager.submitTrigger(
                     TriggerProfile.parseJsonStr(jobProfile.toJsonStr()));
-            } else if (jobProfile.hasKey(JOB_DIR_FILTER_PATTERN)) {
-                jobManager.submitFileJobProfile(jobProfile);
             } else {
-                jobManager.submitSqlJobProfile(jobProfile);
+                TaskTypeEnum taskType = TaskTypeEnum
+                    .getTaskType(jobProfile.getInt(JOB_SOURCE_TYPE));
+                switch (taskType) {
+                    case SQL:
+                        jobManager.submitSqlJobProfile(jobProfile);
+                        break;
+                    case FILE:
+                        jobManager.submitFileJobProfile(jobProfile);
+                        break;
+                    case KAFKA:
+                    case BINLOG:
+                        jobManager.submitJobProfile(jobProfile, true);
+                        break;
+                    default:
```

Comment:
the default value better use `UNSUPPORT_TYPE` to mark it, it may be useful to debug if the source type has not been supported.

Reply:
Thanks for the advice